src/reasoning/reasoning_engine.py

```diff
@@ -0,0 +1,71 @@
+from typing import Dict, List, Tuple
+
+from src.data.generate_descriptor_qas import (
+    generate_descriptor_core_question,
+)
+from src.data.query_item import QueryItem
+from src.models.base_inference import BaseInferenceEngine
+from src.utils.logger import get_logger
+from src.utils.utils import parse_key_objects
+
+logger = get_logger(__name__)
+
+
+class ReasoningEngine:
```

Comment:
Conceptually it is not really clear to me why we dont handle all of this inside of the Dataset processing. This seems like a rather cumbersome way of handling the preparation of the data for training / eval.